### ISSUE TITLE
tiledb_read.py test fails

### ISSUE THREAD

**Issue by rouault** [OSGeo/gdal#1490]
@normanb tiledb_read.py fails for me.
```
$ gdalinfo ../autotest/gdrivers/data/tiledb_array
ERROR 4: `../autotest/gdrivers/data/tiledb_array' not recognized as a supported file format.
gdalinfo failed - unable to open '../autotest/gdrivers/data/tiledb_array'.
```
But doing ```gdal_translate -of TileDB ../autotest/gcore/data/byte.tif tiledb_byte && gdalinfo tiledb_byte``` works

Perhaps a file hasn't been committed ?
```
$ find ../autotest/gdrivers/data/tiledb_array/
../autotest/gdrivers/data/tiledb_array/
../autotest/gdrivers/data/tiledb_array/__lock.tdb
../autotest/gdrivers/data/tiledb_array/test.tdb.aux.xml
../autotest/gdrivers/data/tiledb_array/__array_schema.tdb
../autotest/gdrivers/data/tiledb_array/__d3a25c5c414a4c0eb48b6c12c7a82bb9_1556324032768
../autotest/gdrivers/data/tiledb_array/__d3a25c5c414a4c0eb48b6c12c7a82bb9_1556324032768/__fragment_metadata.tdb
../autotest/gdrivers/data/tiledb_array/__d3a25c5c414a4c0eb48b6c12c7a82bb9_1556324032768/TDB_VALUES.tdb
```

**Comment by normanb** [comment]
I can confirm the same error. I will recreate the array and push a PR. This is related to the changes I just made I suspect with re-ordering the internal axes.

**Comment by rouault** [comment]
@normanb Reopening... I've just merged your PR and pulled and I still get the same error

**Comment by normanb** [comment]
I did see a similar issue with PDAL integration. The default compression in TileDB is zstd and even with zero compression set it is passed through that filter. ZSTD frames are dependent on the library version. I will change the compression on the test array to be gzip (which has backwards compatibility)

**Comment by rouault** [comment]
Ah, I'm indeed using zstd 0.5.1 from Ubuntu 16.04. A bit ancient probably

**Comment by rouault** [comment]
Fix confirmed, and backported to release/2.5
